commonItems/BufferedReader.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace commonItems {
+    public class BufferedReader : StreamReader {
+        // http://web.archive.org/web/20210702221522/https://stackoverflow.com/questions/7049401/c-sharp-roll-back-streamreader-1-character/7050430#7050430
```

Comment:
Move this comment before the class definition begins.

Also flesh it out. Mention that this is a derivation that allows for returning one character earlier in a stream. You may not need the stack overflow link once writing it out, but feel free to still include it.